doc/nrf/protocols/thread/device_types.rst

```diff
@@ -0,0 +1,84 @@
+.. _thread_device_types:
+
+Thread device types
+###################
+
+.. contents::
+   :local:
+   :depth: 2
+
+Thread devices can be either Full Thread Devices (FTD) or Minimal End Devices (MTD).
+The devices are divided based on their ability exist in a network without a parent, determined by whether the device maintains a routing table (doesn't need a parent) or not (needs a parent).
+FTDs are the devices that maintain a routing table, while MTDs do not, forwarding all their messages to a parent instead.
+
+Another type separation is based on the device's ability to route messages.
+We can divide thread devices into Routers and End Devices: Routers can route messages, while End Devices cannot.
+FTDs can be either Routers or End Devices, while MTDs can only be End Devices.
+Only Routers can function as parents.
+
+For additional information about Thread device types, see the documentation for `Device Types on OpenThread portal`_.
+
+.. _thread_types_ftd:
+
+Full Thread Devices
+*******************
+
+In order to maintain the routing table, FTDs keep their radio on at all times.
+An important difference between FTDs and MTDs is that FTDs subscribe to special "all-routers" multicast addresses (``ff02::2`` and ``ff03::2``, see `IPv6 multicast addressing`_).
+Because of these reasons, they typically consume more power than MTDs.
+
+Full Thread Devices can be further divided into three categories:
+
+* Router
+* Router Eligible End Device (REED)
+* Full End Device (FED)
+
+Routers can be regarded as the backbone of a Thread network.
+They maintain the routing table and forward messages to other devices.
+A router is required for a Minimal End Device to join the network.
+Routers, unlike End Devices, can also be used to extend the network range.
+
+Both Routers and REEDs have routing capabilities, while FEDs do not.
+REEDs function similar to FEDs, but they can become Routers if needed.
+If a REED is the only device in range of an End Device trying to join the network it will promote itself to a Router (see `Joining an existing Thread network`_ in the OpenThread documentation for more information).
+Conversely, when a Router has no children it can downgrade itself to a REED.
```

Comment:
I am now thinking we could add a sentence informing that the limit of routers is 32 for a single network (enforced by 5-bit router id encoding in RLOC16)

Reply:
Added mention of maximum number, but left the bit about RLOC16 out.